src/renderer/controllers/subtitles-controller.js

```diff
@@ -137,3 +155,49 @@ function relabelSubtitles (subtitles) {
     track.label = counts[lang] > 1 ? (lang + ' ' + counts[lang]) : lang
   })
 }
+
+function detectVTTLanguage (buffer) {
+  var LanguageDetect = require('languagedetect')
+
+  // Detect what language the subtitles are in
+  var vttContents = buffer.toString().replace(/(.*-->.*)/g, '') // remove numbers?
+  var langDetected = (new LanguageDetect()).detect(vttContents, 2)
+  langDetected = langDetected.length ? langDetected[0][0] : 'subtitle'
+  langDetected = langDetected.slice(0, 1).toUpperCase() + langDetected.slice(1)
+
+  return langDetected
+}
+
+function loadEmbeddedSubtitle (trackEntry) {
+  // convert to .vtt format
+  var vtt = 'WEBVTT FILE\r\n\r\n'
+  trackEntry.subtitles.forEach(function (sub, i) {
+    vtt += `${i + 1}\r\n`
+    vtt += `${msToTime(sub.time)} --> ${msToTime(sub.time + sub.duration)}\r\n`
+    vtt += `${sub.text}\r\n\r\n`
+  })
+
+  function msToTime (s) {
+    var ms = s % 1000
+    s = (s - ms) / 1000
+    var secs = s % 60
+    s = (s - secs) / 60
+    var mins = s % 60
+    var hrs = (s - mins) / 60
+
+    var z = zeroFill
+    return z(2, hrs) + ':' + z(2, mins) + ':' + z(2, secs) + '.' + z(3, ms)
+  }
+
+  var buf = new Buffer(vtt)
+  var langDetected = detectVTTLanguage(buf)
+
+  var track = {
+    buffer: 'data:text/vtt;base64,' + buf.toString('base64'),
+    language: langDetected,
+    label: langDetected,
+    filePath: null
```

Comment:
The `filePath` is used to persist the subtitle that is selected. See `player-page.js` for where it's saved. And in `playback-controller.js` it's read out as `fileSummary.selectedSubtitle`.

We should change this so it works for subtitles embedded within mkv files too.